azure-proto-core/Resources/IdentityKind.cs

```diff
@@ -20,22 +18,52 @@ public IdentityKind(string kind)
         public string Value { get; private set; }
         public int CompareTo(IdentityKind other)
         {
-            throw new NotImplementedException();
+            bool? checkNull = checkBothNull(other.Value);
+            if (checkNull == true)
+                return 0;
+            else if (checkNull == false)
+                return -1;
+            return this.Value.CompareTo(other.Value);
         }
 
         public int CompareTo(string other)
         {
-            throw new NotImplementedException();
+            bool? checkNull = checkBothNull(other);
+            if (checkNull == true)
+                return 0;
+            else if (checkNull == false)
+                return -1;
+            return this.Value.CompareTo(other);
         }
 
         public bool Equals(IdentityKind other)
         {
-            throw new NotImplementedException();
+            bool? checkNull = checkBothNull(other.Value);
+            if (checkNull == true)
+                return true;
+            if (checkNull == false)
+                return false;
+            return this.Value.Equals(other.Value);
         }
 
         public bool Equals(string other)
         {
-            throw new NotImplementedException();
+            bool? checkNull = checkBothNull(other);
+            if (checkNull == true)
+                return true;
+            if (checkNull == false)
+                return false;
+            return this.Value.Equals(other);
+        }
+
+        private bool? checkBothNull(string other)
+        {
+            bool? result = null;
+            if (this.Value == null && other == null)
+                result = true;
+            else if (this.Value == null)
+                result = false;
+            return result;
         }
     }
-}
+}
```

Comment:
add newline for file

Reply:
Where would I put the newline? Should I put an enter at line 70?